### ISSUE TITLE
Amazon ElastiCache for Redis adds support for customer managed keys

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#9939]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon ElastiCache for Redis now supports encryption at rest using customer managed customer master keys (CMKs) in AWS Key Management Service (KMS).

### New or Affected Resource(s)

* [`aws_elasticache_replication_group`](https://www.terraform.io/docs/providers/aws/r/elasticache_replication_group.html)

### Potential Terraform Configuration

```hcl
resource "aws_elasticache_replication_group" "example" {
  kms_key_id = "${aws_kms_key.example.arn}"
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/08/amazon-elasticache-for-redis-adds-support-for-customer-managed-keys-in-aws-key-management-service-for-encryption-at-rest/).
[API reference](https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateReplicationGroup.html).

**Comment by praveensastry** [comment]
@ewbankkit Are you going to open a PR for this? I'd be willing to do so if this is not a top priority for you now.

**Comment by ewbankkit** [comment]
@praveensastry Please go ahead.

**Comment by kingslyt** [comment]
@praveensastry  Did you get a chance to work on this?

**Comment by TarunParwani** [comment]
guys any update on this issue? is this now supported?

**Comment by turaleck** [comment]
Any update on this issue?

**Comment by bflad** [comment]
This has been released in [version 2.34.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2340-october-31-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
